### PR TITLE
Set-BqTable + Tests

### PR DESCRIPTION
**Pull request by ahandley**

### CODE REVIEW

File: Google.PowerShell/BigQuery/BqTable.cs [hunk 1]
```diff
@@ -163,6 +163,63 @@ public void DoGetRequest(TablesResource.GetRequest request)
 
     /// <summary>
     /// <para type="synopsis">
+    /// Updates information describing an existing BigQuery table.
+    /// </para>
+    /// <para type="description">
+    /// text
+    /// </para>
+    /// <example>
+    ///   <code>PS C:\> $my_tab = Get-BqTable -DatasetId “my_data” “my_table”
```
Comment (ILMTitan) [inline Google.PowerShell/BigQuery/BqTable.cs]: For multi-line examples like this, give the start and end tags their own lines, and indent properly. `<code>` preserves whitespace, except it also trims down to relevant text.  Just keep consistent indentation, and don't worry about empty starting and trailing newlines. Follow normal rules for how much to indent, and have nice alignment within that indent.

File: Google.PowerShell/BigQuery/BqTable.cs [hunk 2]
```diff
@@ -163,6 +163,63 @@ public void DoGetRequest(TablesResource.GetRequest request)
...
+        protected override void ProcessRecord()
+        {
+            Table response;
+            var request = Service.Tables.Update(InputObject,
+                InputObject.TableReference.ProjectId,
+                InputObject.TableReference.DatasetId,
+                InputObject.TableReference.TableId);
+            try
+            {
+                response = request.Execute();
```
Comment (ILMTitan) [inline Google.PowerShell/BigQuery/BqTable.cs]: You should write the updated table to the object pipeline.

File: Google.PowerShell.IntegrationTests/BigQuery/BigQuery.BqTable.Tests.ps1 [hunk 3]
```diff
@@ -63,6 +63,76 @@ Describe "Get-BqTable" {
...
+
+    BeforeAll {
+        $r = Get-Random
+        $datasetName = "pshell_testing_$r"
+        $test_set = New-BqDataset $datasetName
+    }
+
+    It "should update trivial metadata fields via pipeline" {
+        try {
+            New-BqTable -Dataset $test_set "pipe_table" -Name "test" -Description "my table"
```
Comment (ahandley) [inline Google.PowerShell.IntegrationTests/BigQuery/BigQuery.BqTable.Tests.ps1]: The point of this is the server copy has different values, so when we run Set- for the first time, they're updated.

Normally yes.  A user would just put the correct values into the New- command in the first place.

File: Google.PowerShell.IntegrationTests/BigQuery/BigQuery.BqTable.Tests.ps1 [hunk 4]
```diff
@@ -63,6 +63,76 @@ Describe "Get-BqTable" {
...
+            $table.FriendlyName = "Some Test Data"
+            $table.Description = "A new description!"
+            $table | Set-BqTable
+
+            $table = Get-BqTable -Dataset $test_set "pipe_table"
+            $table | Should Not BeNullOrEmpty
+            $table.FriendlyName | Should Be "Some Test Data"
+            $table.Description | Should Be "A new description!"
+        }
+        finally {
```
Comment (ahandley) [inline Google.PowerShell.IntegrationTests/BigQuery/BigQuery.BqTable.Tests.ps1]: Fixed all instances.

File: Google.PowerShell/BigQuery/BqTable.cs [hunk 5]
```diff
@@ -163,6 +163,63 @@ public void DoGetRequest(TablesResource.GetRequest request)
 
     /// <summary>
     /// <para type="synopsis">
+    /// Updates information describing an existing BigQuery table.
+    /// </para>
+    /// <para type="description">
+    /// text
```
Comment (ahandley) [inline Google.PowerShell/BigQuery/BqTable.cs]: Agreed.  The description is not so useful when the coder forgets to copy-paste it out of the design doc.

File: Google.PowerShell/BigQuery/BqTable.cs (hunk 1 above)
Comment (ahandley) [inline Google.PowerShell/BigQuery/BqTable.cs]: Fixed all instances.  Thanks!

File: Google.PowerShell/BigQuery/BqTable.cs (hunk 2 above)
Comment (ahandley) [inline Google.PowerShell/BigQuery/BqTable.cs]: Done.

File: Google.PowerShell/BigQuery/BqDataset.cs [hunk 6]
```diff
@@ -272,8 +276,10 @@ protected override void ProcessRecord()
     ///   <para>This takes a Dataset object from the pipeline and inserts it into the Cloud project "my-project".</para>
     /// </example>
     /// <example>
-    ///   <code>PS C:\> New-BqDataset "test_data_id" -Name "Testdata" `
-    ///   -Description "Some interesting data!" -Expiration 86400000</code>
+    ///   <code>
+    /// PS C:\> New-BqDataset "test_data_id" -Name "Testdata" `
+    /// -Description "Some interesting data!" -Expiration 86400000
```
Comment (ILMTitan) [inline Google.PowerShell/BigQuery/BqDataset.cs]: This line should be indented, so it looks more like

    PS C:> New-BqDataset "test_data_id" ...
                         -Description ...

You may want to add one more line break if you do so.

File: Google.PowerShell/BigQuery/BqTable.cs [hunk 7]
```diff
@@ -231,8 +235,10 @@ protected override void ProcessRecord()
     /// a Table object.
     /// </para>
     /// <example>
-    ///   <code>PS C:\> New-BqTable “new_tab” -Dataset “my_data” -Description “Some nice data!” `
-    ///   -Expiration (60*60*24*30)</code>
+    ///   <code>
+    /// PS C:\> New-BqTable “new_tab” -Dataset “my_data” -Description “Some nice data!” `
+    /// -Expiration (60*60*24*30)
```
Comment (ILMTitan) [inline Google.PowerShell/BigQuery/BqTable.cs]: Align `-Expiration` closer to the parameters.

File: Google.PowerShell/BigQuery/BqDataset.cs (hunk 6 above)
Comment (ahandley) [inline Google.PowerShell/BigQuery/BqDataset.cs]: Got em!  Thanks!

**Pull request merged**